python/langsmith/wrappers/_openai.py

```diff
@@ -160,6 +160,35 @@ def _reduce_completions(all_chunks: List[Completion]) -> dict:
     return d
 
 
+_DICT: Dict[str, Any] = {}
+
+
+def _flatten_metadata(metadata: Optional[dict]):
+    if not isinstance(metadata, dict):
+        return {}
+    metadata["input_tokens"] = metadata.pop("prompt_tokens", None)
+    metadata["output_tokens"] = metadata.pop("completion_tokens", None)
+    metadata["input_token_details"] = {
+        k.rstrip("_tokens"): v
+        for k, v in (metadata.get("prompt_tokens_details") or _DICT).items()
+    }
+    metadata["output_tokens_details"] = {
+        k.rstrip("_tokens"): v
+        for k, v in (metadata.get("completion_tokens_details") or _DICT).items()
+    }
+    return metadata
+
+
+def _count_beans(outputs: Any):
+    try:
+        rdict = outputs.model_dump()
+        rdict["usage_metadata"] = _flatten_metadata(rdict.pop("usage", None))
+        return rdict
```

Comment:
just flagging that this will have the potentially unintended consequence of making the output a dict instead of a `ChatCompletion` object -- the output will be inlined in `outputs` as opposed to nested under `outputs.output`

Reply:
Just in the run.outputs right? (not in the actual application response)

That's what we do in the backend anyway, right?